modules/eco-configuring-control-plane-machine-health-check-with-self-node-remediation.adoc

```diff
@@ -0,0 +1,88 @@
+// Module included in the following assemblies:
+//
+// *nodes/nodes/eco-poison-pill-operator.adoc
+
+:_content-type: PROCEDURE
+[id="configuring-control-plane-machine-health-check-with-self-node-remediation-operator_{context}"]
+= Configuring control-plane machine health checks to use the Self Node Remediation Operator
+
+Use the following procedure to configure the control-plane machine health checks to use the Self Node Remediation Operator as a remediation provider.
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
+* Log in as a user with `cluster-admin` privileges.
+
+.Procedure
+
+. Create a `SelfNodeRemediationTemplate` CR:
+
+.. Define the `SelfNodeRemediationTemplate` CR:
++
+[source,yaml]
+----
+apiVersion: self-node-remediation.medik8s.io/v1alpha1
+kind: SelfNodeRemediationTemplate
+metadata:
+  namespace: openshift-machine-api
+  name: selfnoderemediationtemplate-sample
+spec:
+  template:
+    spec:
+      remediationStrategy: ResourceDeletion <1>
+----
+<1> Specifies the remediation strategy. The default strategy is `ResourceDeletion`.
+
+.. To create the `SelfNodeRemediationTemplate` CR, run the following command:
++
+[source,terminal]
+----
+$ oc create -f <snr-name>.yaml
+----
+
+. Create or update the `MachineHealthCheck` CR to point to the `SelfNodeRemediationTemplate` CR:
+
+.. Define or update the `MachineHealthCheck` CR:
++
+[source,yaml]
+----
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineHealthCheck
+metadata:
+  name: machine-health-check
+  namespace: openshift-machine-api
+spec:
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-machine-role: "control-plane"
+      machine.openshift.io/cluster-api-machine-type: "control-plane"
+  unhealthyConditions:
+  - type:    "Ready"
+    timeout: "300s"
+    status: "False"
+  - type:    "Ready"
+    timeout: "300s"
+    status: "Unknown"
+  maxUnhealthy: "40%"
+  nodeStartupTimeout: "10m"
+  remediationTemplate: <1>
+    kind: SelfNodeRemediationTemplate
+    apiVersion: self-node-remediation.medik8s.io/v1alpha1
+    name: selfnoderemediationtemplate-sample
+----
+<1> Specifies the details for the remediation template.
+
++
+.. To create a `MachineHealthCheck` CR, run the following command:
++
+[source,terminal]
+----
+$ oc create -f <file-name>.yaml
```

Comment:
maybe better: "mhc-name" instead of "file-name"

Reply:
Thanks for your feedback Polina. I have applied this change.